### PR TITLE
allow $ escaping double quotes in interpolations

### PR DESCRIPTION
**Pull request by martijnhoekstra**
As suggested by @som-snytt, making compatibility between scala2 and scala3 easier.

cf.
  - https://github.com/lampepfl/dotty/pull/7486
  - https://contributors.scala-lang.org/t/pre-sip-mini-sip-escapes-in-interpolations/2076

### CODE REVIEW

**Comment by dwijnand** [comment]
Gonna merge conflict with #8830.

File: src/compiler/scala/tools/nsc/ast/parser/Scanners.scala [hunk 1]
```diff
@@ -912,7 +912,7 @@ trait Scanners extends ScannersCommon {
         }
       } else if (ch == '$') {
         nextRawChar()
-        if (ch == '$') {
+        if (ch == '$' || (currentRun.isScala3 && ch == '"')) {
```
Comment (lrytz) [inline src/compiler/scala/tools/nsc/ast/parser/Scanners.scala]: IMO we should allow this in 2.13, it doesn't break any existing code.

Comment (martijnhoekstra) [reply src/compiler/scala/tools/nsc/ast/parser/Scanners.scala]: That's still in pre-SIP state: https://contributors.scala-lang.org/t/pre-sip-mini-sip-escapes-in-interpolations/2076/

Comment (lrytz) [reply src/compiler/scala/tools/nsc/ast/parser/Scanners.scala]: Yeah, but it's in Scala 3, so 🤷‍♂️

Comment (martijnhoekstra) [reply src/compiler/scala/tools/nsc/ast/parser/Scanners.scala]: Then it probably also needs the spec changes in https://github.com/scala/scala/pull/6953

**Review by lrytz — APPROVED** [review]
Thank you!

**Pull request merged**